src/registries/npm-registry.js

```diff
@@ -178,6 +177,12 @@ export default class NpmRegistry extends Registry {
       [false, path.join(getGlobalPrefix(), 'etc', filename)],
     ];
 
+    // When home directory for global install is different from where $HOME/npmrc is stored,
+    // E.g. /usr/local/share vs /root on linux machines, check the additional location
+    if (home !== userHome) {
+      possibles.push([true, path.join(home, localfile)]);
```

Comment:
I think we should change this data structure to be a normal object instead of an array sometime in the future: `{isHomeDir: true, path: path.join(home, localfile)}`